### ISSUE TITLE
add configuration option to hide DSpace version number

### ISSUE THREAD

**Issue by saschaszott** [DSpace/DSpace#9780]
## Change Request

The (publicly available) REST endpoint `/server/api` reveals the number of the deployed DSpace version, e.g.

```
$ curl -sS 'https://demo.dspace.org/server/api' 2>&1 | grep 'dspaceVersion'
"dspaceVersion" : "DSpace 8.1-SNAPSHOT",
```

We propose to add a new configuration option (in `dspace.cfg`) that allows instances to suppress the output of the version number (e.g., due to security concerns).

**Comment by tdonohue** [comment]
I disagree with this request, because this version number is often necessary for any support requests.  One of the first pieces of information we ask for is the version number, and making the version more difficult to locate will make support more difficult.

The version is also necessary for updates to any public registries of known DSpace sites.  It also may be necessary for public crawlers sometimes (Google Scholar) to ensure it can accurately crawl your site.

While I understand the argument that it could be used to determine if your site *might be* vulnerable to certain attacks, the version shared is never a direct indication of what patches you have applied.  So, a version reported as "7.6.1" may not be vulnerable to issues fixed in 7.6.2 if those fixes were applied via patches.

If we felt that sharing a specific version is still an issue, maybe there's a way to hide parts of the version number.  For example, sharing that you are running "7.6.x" or "8.x" may be sufficient for the needs I describe above.   But, I don't think we should hide the version entirely.

**Comment by saschaszott** [comment]
@tdonohue , I don't want to change the default behaviour of DSpace but let instances decide if they want to suppress the version number intentionally by changing the default configuration.

**Comment by saschaszott** [comment]
As an logged-in user with admin privileges you can get the DSpace version by using the Spring info actuator, e.g. 

https://demo.dspace.org/server/#https://demo.dspace.org/server/actuator/info

**Comment by tdonohue** [comment]
@saschaszott : I'd still rather we reframe this as the ability to "anonymize" the version somewhat, by reporting simply "7.x" or "8.x".   I'm still worried that SEO and similar may be impacted if we remove the version entirely (as there will no longer be any way for crawlers to make crawling decisions based on the version of DSpace the crawler encounters).  It also makes it *very difficult* for any automated tools to track of which versions of DSpace are being used in production (e.g. for building a registry of sites).

These are the specific reasons why we chose to make every DSpace share it's version publicly in the HTML `<head>`.  They could likely still be achieved by only reporting the *major* version.  However, allowing sites to _easily_ hide the version entirely seems like a step backwards.
